packages/core/src/storages/disk-storage.ts

```diff
@@ -82,27 +82,26 @@ export class DiskStorage extends BaseStorage<DiskFile> {
     if (file.status === 'completed') return file;
     if (part.size && part.size < file.size) {
       file.size = part.size;
-      await this.saveMeta(file);
     }
     if (!isValidPart(part, file)) return fail(ERRORS.FILE_CONFLICT);
     try {
       file.bytesWritten = await this._write({ ...file, ...part });
       if (file.bytesWritten === INVALID_OFFSET) return fail(ERRORS.FILE_CONFLICT);
       file.status = getFileStatus(file);
-      if (file.status === 'completed') await this.saveMeta(file);
+      await this.saveMeta(file);
       return file;
     } catch (err) {
       return fail(ERRORS.FILE_ERROR, err);
     }
   }
 
   async delete({ id }: FilePart): Promise<DiskFile[]> {
-    const file = await this.getMeta(id).catch(() => null);
-    if (file) {
-      await removeFile(this.getFilePath(file)).catch(() => null);
+    try {
+      const file = await this.getMeta(id);
+      await removeFile(this.getFilePath(file));
```

Comment:
Is it possible to `catch` this call? We may have a META file but not the physical file (moved after a successful upload for example).

Reply:
Thanks! #527

Reply:
How about #528?

Reply:
Maybe #528 ?

Reply:
> How about https://github.com/kukhariev/node-uploadx/pull/528?

Seem good, thanks!